lib/Runtime/Language/JavascriptExceptionOperators.cpp

```diff
@@ -238,6 +280,53 @@ namespace Js
         {
             // Clone static exception object early in case finally block overwrites it
             exception = exception->CloneIfStaticExceptionObject(scriptContext);
+
```

Comment:
you need to check for hasBailedOutOffset in this function and also generate code in LowerMD.cpp to pass the param to this helper